File: 3555-final-array-state-after-k-multiplication-operations-i/solution.py
class Solution:
    def getFinalState(self, nums: List[int], k: int, multiplier: int) -> List[int]:
        pq = [(val, i) for i, val in enumerate(nums)] # priority queue or min heap
        heapify(pq) # min heap done

        for _ in range(k):
            _, i = heappop(pq) # _ is val (not needed), i is index (needed)
            nums[i] *= multiplier
            heappush(pq, (nums[i], i))
        
        return nums

File: 3555-final-array-state-after-k-multiplication-operations-i/solution.cpp
class Solution {
public:
    vector<int> getFinalState(vector<int>& nums, int k, int multiplier) {
        int n = nums.size();

        while (k--) {
            int minidx = 0;
            for (int i = 0; i < n; i++) {
                if (nums[i] < nums[minidx]) {
                    minidx = i;
                }
            }

            nums[minidx] *= multiplier;
        }

        return nums;
    }
};
